Classes/Player/PlayerController.swift

```diff
@@ -166,25 +174,24 @@ extension PlayerController {
 extension PlayerController {
     
     private func shouldRefreshAsset() {
-        if let handler = self.assetBuilder?.assetHandler as? RefreshableAssetHandler {
-            if let (source, _) = self.assetBuilder!.getPreferredMediaSource() {
-                handler.shouldRefreshAsset(mediaSource: source) { [unowned self] (shouldRefresh) in
-                    if shouldRefresh {
-                        self.shouldRefresh = true
-                    }
-                }
+        guard let mediaEntry = self.mediaEntry,
```

Comment:
Something here (together with `refreshAsset()` below) doesn't make sense. Both functions call `AssetBuilder.getPreferredMediaSource`. Not even sure why -- it will not change just because we're back from background or something. 

Also, this function does not return Bool, so it should not be called "shouldXYZ". It should be called something like "checkIfAssetShouldRefresh" or at least "checkShouldRefreshAsset".